### PR TITLE
zh-cn: init the translation of svelte todo list begin

### PR DESCRIPTION
**Pull request by hanyujie2002**
### Description

Same as title
### Motivation

Ditto
### Additional details

Based on the [zh-TW translation](https://github.com/mdn/translated-content/blob/main/files/zh-tw/learn/tools_and_testing/client-side_javascript_frameworks/svelte_todo_list_beginning/index.md)
### Related issues and pull requests

Relates to #16965

### CODE REVIEW

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_todo_list_beginning/index.md [hunk 1]
```diff
@@ -0,0 +1,720 @@
...
+```svelte
+<ul
+  role="list"
+  className="todo-list stack-large"
+  aria-labelledby="list-heading"></ul>
+```
+
+`role` 属性则帮助无障碍技术去解释元素具有什么种类的语义数值——或它的目的。默认情况下，`<ul>` 会被视为列表，但在加上样式之后会破坏原有性质。明确定义规则为“list”将能复原 `<ul>` 元素本身意义。假如你想要知道更多为什么需要明确定义的话，可以去参考 Scott O'Hara 的[“修复列表”](https://www.scottohara.me/blog/2019/01/12/lists-and-safari.html)（2019）文章。
+
+`aria-labelledby` 属性则告诉辅助技术，我们把 `<h2>` 且 `id` 为 `list-heading` 作为描述下面列表用途的标签。建立这种关联为列表提供更多上下文信息，这可以帮助屏幕阅读器用户更好地去了解它的目的。
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_todo_list_beginning/index.md:303]:
```suggestion
`aria-labelledby` 属性则告诉辅助技术，我们把具有 `id` 属性值为 `list-heading` 的 `<h2>` 元素作为描述下面列表用途的标签。这种关联为列表提供了更多的上下文信息，可以帮助屏幕阅读器用户更好地理解列表的目的。
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_todo_list_beginning/index.md [hunk 2]
```diff
@@ -0,0 +1,720 @@
...
+
+这似乎是讨论 Svelte 如何处理无障碍的好时机；让我们继续看下去。
+
+## Svelte 无障碍性支持
+
+Svelte 特别重视无障碍性。目的是鼓励开发人员尽可能预设地写出无障碍程序代码。作为编译程序，Svelte 能静态地分析我们的 HTML 模板，而当组件被编译后，就能提出一些无障碍警告。
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_todo_list_beginning/index.md:309]:
```suggestion
Svelte 特别重视无障碍性。目的是鼓励开发人员尽可能“默认”编写更具有无障碍性的代码。作为编译程序，Svelte 能静态地分析我们的 HTML 模板，而当组件被编译后，就能提出一些无障碍警告。
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_todo_list_beginning/index.md [hunk 3]
```diff
@@ -0,0 +1,720 @@
...
+
+无障碍性（缩写为 a11y）一直以来不容易写正确，但 Svelte 将会协助并警告你，假如你写出了不容易阅读的标记语言。
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_todo_list_beginning/index.md:311]:
```suggestion
无障碍性（缩写为 a11y）并不总是容易做到，但 Svelte 会在编写不具无障碍性的标记时发出警告，从而帮助你。
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_todo_list_beginning/index.md [hunk 4]
```diff
@@ -0,0 +1,720 @@
...
+
+举例来说，假如我们新增 `<img>` 元素至我们的 `todos.svelte` 组件，但没有对应的 `alt` 属性：
+
+```html
+<h1>Svelte To-Do list</h1>
+
+<img height="32" width="88" src="https://www.w3.org/WAI/wcag2A" />
+```
+
+编译程序将会报出下列警告：
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_todo_list_beginning/index.md:321]:
```suggestion
编译器将会报出下列警告：
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_todo_list_beginning/index.md [hunk 5]
```diff
@@ -0,0 +1,720 @@
...
+1: <h1>Svelte To-Do list</h1>
+2:
+3: <img height="32" width="88" src="https://www.w3.org/WAI/wcag2A">
+   ^
+created public/build/bundle.js in 220ms
+
+[2020-07-15 04:07:43] waiting for changes...
+```
+
+此外，在我们调用编译程序之前，编辑器也会显示这个警告：
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_todo_list_beginning/index.md:335]:
```suggestion
此外，在我们调用编译器之前，编辑器也会显示这个警告：
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_todo_list_beginning/index.md [hunk 6]
```diff
@@ -0,0 +1,720 @@
...
+
+![显示图像标签的代码编辑器窗口，并弹出错误消息，指出该元素应具有 alt 属性](04-svelte-accessibility-support.png)
+
+你可以使用以 `svelte-ignore` 开头的[批注](https://svelte.dev/docs#Comments)，来告诉 Svelte 去忽略下一个标记语言区块的警告，像是如下：
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_todo_list_beginning/index.md:339]:
```suggestion
你可以像这样使用以 `svelte-ignore` 开头的[注释](https://svelte.dev/docs#Comments)，来告诉 Svelte 去忽略下一个标记语言区块的警告：
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_todo_list_beginning/index.md [hunk 7]
```diff
@@ -0,0 +1,720 @@
...
+
+```svelte
+<!-- svelte-ignore a11y-missing-attribute -->
+<img height="32" width="88" src="https://www.w3.org/WAI/wcag2A" />
+```
+
+> **备注：** 透过 VSCode 你可以点击 _Quick fix..._ 链接或按 <kbd>Ctrl</kbd> + <kbd>.</kbd> 来自动地加上这个忽略批注。
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_todo_list_beginning/index.md:346]:
```suggestion
> **备注：** 透过 VSCode 你可以点击 _Quick fix..._ 链接或按 <kbd>Ctrl</kbd> + <kbd>.</kbd> 来自动地加上这个忽略注释。
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_todo_list_beginning/index.md [hunk 8]
```diff
@@ -0,0 +1,720 @@
...
+
+假如你想要全局地关闭这个警告，可以在你的 `rollup.config.js` 文件中调整 `Svelte` 插件并补上这个 `onwarn` 处理程序即可，像是如下：
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_todo_list_beginning/index.md:348]:
```suggestion
假如你想要全局地关闭这个警告，可以像这样在你的 `rollup.config.js` 文件中调整 `Svelte` 插件配置并补上这个 `onwarn` 处理器：
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_todo_list_beginning/index.md [hunk 9]
```diff
@@ -0,0 +1,720 @@
...
+      // 让 Rollup 正常处理所有其他警告
+      handler(warning);
+    },
+  }),
+
+  // …
+];
+```
+
+在设计上，这些警告被实现于编译程序本身，而不是你可以选择性加入项目的插件。理念是默认会检查你的标记语言有无 a11y 问题并让你可以选择排除特定警告。
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_todo_list_beginning/index.md:373]:
```suggestion
在设计上，这些警告是在编译器本身中实现的，而不是可以选择性加入项目的插件。这样做的目的是在默认情况下检查标记中的无障碍问题，并让你选择不使用特定的警告。
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_todo_list_beginning/index.md [hunk 10]
```diff
@@ -0,0 +1,720 @@
...
+
+> **备注：** 假如你有足够充分的理由，则可以先停用这些警告，例如当你在快速建构原型时。但请记住你需要成为优秀的网民并让你的页面尽可能无障碍，可以更广泛地被用户阅读。
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_todo_list_beginning/index.md:375]:
```suggestion
> **备注：** 只有在有充分理由的情况下，例如在构建快速原型时，才应禁用这些警告。做一名优秀的网络公民，让尽可能多的用户都能访问你的网页，这一点非常重要。
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_todo_list_beginning/index.md [hunk 11]
```diff
@@ -0,0 +1,720 @@
...
+
+Svelte 透过 [eslint-plugin-jsx-a11y](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y#supported-rules) 来检查无障碍性规则，它是提供在 JSX 元素方面的许多无障碍性规则静态检查的一个 ESLint 插件。Svelte 企图将它们全部实现于编译程序中，而大部分都已经移植至 Svelte 中。你可以在 GitHub 中看到[尚未实现的无障碍检查](https://github.com/sveltejs/svelte/issues/820)。点击规则的链接可以查阅它们各自的涵义。
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_todo_list_beginning/index.md:377]:
```suggestion
Svelte 通过 [eslint-plugin-jsx-a11y](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y#supported-rules) 来检查无障碍性规则，它是提供在 JSX 元素方面的许多无障碍性规则静态检查的一个 ESLint 插件。Svelte 的目标是在其编译器中实现所有这些功能，其中大部分功能已经移植到 Svelte 中。你可以在 GitHub 中看到[尚未实现的无障碍检查](https://github.com/sveltejs/svelte/issues/820)。点击规则的链接可以查阅它们各自的含义。
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_todo_list_beginning/index.md [hunk 12]
```diff
@@ -0,0 +1,720 @@
...
+
+## 为标记语言加上样式
+
+让我们来将待办列表变得好看些。将以下内容取代至 `public/global.css` 文件中：
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_todo_list_beginning/index.md:381]:
```suggestion
让我们来将待办列表变得好看些。将 `public/global.css` 文件替换为如下内容：
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_todo_list_beginning/index.md [hunk 13]
```diff
@@ -0,0 +1,720 @@
...
+
+```css
+/* 重置 */
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_todo_list_beginning/index.md:384]:
```suggestion
/* 重置样式 */
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_todo_list_beginning/index.md [hunk 14]
```diff
@@ -0,0 +1,720 @@
...
+  background-color: #f5f5f5;
+  color: #4d4d4d;
+}
+@media screen and (min-width: 620px) {
+  body {
+    font-size: 1.9rem;
+    line-height: 1.31579;
+  }
+}
+/* 重置结尾 */
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_todo_list_beginning/index.md:455]:
```suggestion
/* 重置样式结尾 */
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_todo_list_beginning/index.md [hunk 15]
```diff
@@ -0,0 +1,720 @@
...
+
+```bash
+npx degit opensas/mdn-svelte-tutorial/03-adding-dynamic-behavior
+```
+
+记得执行 `npm install && npm run dev` 以开发模式来运行你的应用程序。
+
+### REPL
+
+若要在 REPL 看见当前程序代码状态，请点击如下连结：
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_todo_list_beginning/index.md:712]:
```suggestion
若要在 REPL 看见当前程序代码状态，请点击如下链接：
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_todo_list_beginning/index.md [hunk 16]
```diff
@@ -0,0 +1,720 @@
...
+
+<https://svelte.dev/repl/c862d964d48d473ca63ab91709a0a5a0?version=3.23.2>
+
+## 总结
+
+随着我们对标记语言加上样式后，待办列表应用程序开始逐渐成形，我们终于可以专注在需要实现的功能上了。
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_todo_list_beginning/index.md:718]:
```suggestion
随着标记和样式设计的到位，我们的待办事项列表应用程序已初具雏形，一切准备就绪，我们可以开始专注于需要实现的特性了。
```

**Review by jasonren0403 — CHANGES_REQUESTED** [review]
Please, pay attention to the language and grammar

**Review by jasonren0403 — APPROVED** [review]
Thanks @hanyujie2002 
One final thing, could we add [`l10n.sourceCommit`](https://github.com/mdn/translated-content/blob/main/docs/zh-cn/translation-guide.md#%E5%85%83%E6%95%B0%E6%8D%AE) in the metadata field? We may start tracking them in the future.

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_todo_list_beginning/index.md [hunk 17]
```diff
@@ -0,0 +1,719 @@
+---
+title: 开始编写我们的 Svelte 待办列表应用程序
+slug: Learn/Tools_and_testing/Client-side_JavaScript_frameworks/Svelte_Todo_list_beginning
+---
+
+{{LearnSidebar}}
+{{PreviousMenuNext("Learn/Tools_and_testing/Client-side_JavaScript_frameworks/Svelte_getting_started","Learn/Tools_and_testing/Client-side_JavaScript_frameworks/Svelte_variables_props", "Learn/Tools_and_testing/Client-side_JavaScript_frameworks")}}
+
+现在我们对 Svelte 运作机制有初步的了解后，就能开始建构我们的范例应用程序：一个待办列表。此篇文章中，会先确认应用程序所需的功能，接着会建立 `Todos.svelte` 组件并编写一些静态标记（markup）语言和样式，待一切准备就绪后，就能开始开发待办列表应用程序的相关特性，我们将在后续文章实现它们。
```
Comment (yin1999) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_todo_list_beginning/index.md:9]:
```suggestion
现在我们对 Svelte 运作机制有初步的了解后，就能开始构建我们的示例应用程序：一个待办事项列表。此篇文章中，会先确认应用程序所需的功能，接着会建立 `Todos.svelte` 组件并编写一些静态标记（markup）语言和样式，待一切准备就绪后，就能开始开发待办事项列表应用程序的相关特性，我们将在后续文章实现它们。
```

**Pull request merged**